# 03/05 - Pair Programming Estadística I - Lidia y Tatiana

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

# EJERCICIO 1
**Buscar un conjunto de datos a analizar:**
- **Se recomienda que el conjunto de datos a analizar tenga variables numéricas y categóricas. Ya que tambien se analizarán.**

In [2]:
df= pd.read_csv("./googleplaystore.csv",index_col = 0)
pd.options.display.max_columns=None
df.head()

Category  Rating  \
App                                                                          
Photo Editor & Candy Camera & Grid & ScrapBook      ART_AND_DESIGN     4.1   
Coloring book moana                                 ART_AND_DESIGN     3.9   
U Launcher Lite – FREE Live Cool Themes, Hide Apps  ART_AND_DESIGN     4.7   
Sketch - Draw & Paint                               ART_AND_DESIGN     4.5   
Pixel Draw - Number Art Coloring Book               ART_AND_DESIGN     4.3   

                                                   Reviews  Size     Installs  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook         159   19M      10,000+   
Coloring book moana                                    967   14M     500,000+   
U Launcher Lite – FREE Live Cool Themes, Hide Apps   87510  8.7M   5,000,000+   
Sketch - Draw & Paint                               215644   25M  50,000,000+   
Pixel Draw - Number Art Coloring Book                  967  2.8M     100,000+   

                                                    Type Price Content Rating  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook      Free     0       Everyone   
Coloring book moana                                 Free     0       Everyone   
U Launcher Lite – FREE Live Cool Themes, Hide Apps  Free     0       Everyone   
Sketch - Draw & Paint                               Free     0           Teen   
Pixel Draw - Number Art Coloring Book               Free     0       Everyone   

                                                                       Genres  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook                   Art & Design   
Coloring book moana                                 Art & Design;Pretend Play   
U Launcher Lite – FREE Live Cool Themes, Hide Apps               Art & Design   
Sketch - Draw & Paint                                            Art & Design   
Pixel Draw - Number Art Coloring Book                 Art & Design;Creativity   

                                                        Last Updated  \
App                                                                    
Photo Editor & Candy Camera & Grid & ScrapBook       January 7, 2018   
Coloring book moana                                 January 15, 2018   
U Launcher Lite – FREE Live Cool Themes, Hide Apps    August 1, 2018   
Sketch - Draw & Paint                                   June 8, 2018   
Pixel Draw - Number Art Coloring Book                  June 20, 2018   

                                                           Current Ver  \
App                                                                      
Photo Editor & Candy Camera & Grid & ScrapBook                   1.0.0   
Coloring book moana                                              2.0.0   
U Launcher Lite – FREE Live Cool Themes, Hide Apps               1.2.4   
Sketch - Draw & Paint                               Varies with device   
Pixel Draw - Number Art Coloring Book                              1.1   

                                                     Android Ver  
App                                                               
Photo Editor & Candy Camera & Grid & ScrapBook      4.0.3 and up  
Coloring book moana                                 4.0.3 and up  
U Launcher Lite – FREE Live Cool Themes, Hide Apps  4.0.3 and up  
Sketch - Draw & Paint                                 4.2 and up  
Pixel Draw - Number Art Coloring Book                 4.4 and up

In [3]:
#Queremos ver el número de Columnas y filas de este DATASET
df.shape

(10841, 12)

In [4]:
#Son muchos datos y vamos a valorar sí reducir la información y centrarnos en 3 tipos de APPS
df["Category"].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

In [5]:
#Hemos establecido que el análisis será sobre estos datos, seleccionando 3 categorias para reducir datos.
df_new= df[df["Category"].isin(["DATING", "HEALTH_AND_FITNESS" , "FINANCE"])]


In [6]:
df_new.shape

(941, 12)

In [7]:
#Queremos informarnos del tipo de dato de cada columna
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Index: 941 entries, Moco+ - Chat, Meet People to Gold Quote - Gold.fr
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        941 non-null    object 
 1   Rating          815 non-null    float64
 2   Reviews         941 non-null    object 
 3   Size            941 non-null    object 
 4   Installs        941 non-null    object 
 5   Type            941 non-null    object 
 6   Price           941 non-null    object 
 7   Content Rating  941 non-null    object 
 8   Genres          941 non-null    object 
 9   Last Updated    941 non-null    object 
 10  Current Ver     941 non-null    object 
 11  Android Ver     941 non-null    object 
dtypes: float64(1), object(11)
memory usage: 95.6+ KB


In [8]:
#Vemos que es dtype= object y cambiamos dtype= int64 para poder analizar los datos.

df_new['Reviews'] = df_new['Reviews'].astype('int64')

/tmp/ipykernel_1736/2205491116.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new['Reviews'] = df_new['Reviews'].astype('int64')


In [9]:
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Index: 941 entries, Moco+ - Chat, Meet People to Gold Quote - Gold.fr
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        941 non-null    object 
 1   Rating          815 non-null    float64
 2   Reviews         941 non-null    int64  
 3   Size            941 non-null    object 
 4   Installs        941 non-null    object 
 5   Type            941 non-null    object 
 6   Price           941 non-null    object 
 7   Content Rating  941 non-null    object 
 8   Genres          941 non-null    object 
 9   Last Updated    941 non-null    object 
 10  Current Ver     941 non-null    object 
 11  Android Ver     941 non-null    object 
dtypes: float64(1), int64(1), object(10)
memory usage: 95.6+ KB


# EJERCICIO 2
**Extraer las siguentes estadísticas del conjunto de datos para las varibles disponibles:**

🙋‍♀️ **Las valoraciones y las reseñas tienen un papel muy importante en la visibilidad y el posicionamiento de una app o de un juego móvil en las tiendas de apps, por lo que vamos a estudiar rating y reviews.**

In [10]:
#vamos a definir variables para Reviews y Rating
reviews= df_new["Reviews"]
rating= df_new["Rating"]

>Media

In [11]:
#MEDIA de las Reviews
media_reviews= np.mean(reviews)
print("Media:",media_reviews)

Media: 66669.23379383635


In [12]:
#MEDIA de las Rating
media_rating= np.mean(rating)
print("Media:",media_rating)

Media: 4.146257668711656


>Desviación con respecto de la media

In [13]:
#Desviación con respecto a la media Reviews
desv_media_reviews =np.abs(reviews - media_reviews)
print("Desviación respecto a la media sobre Reviews:",desv_media_reviews)

Desviación respecto a la media sobre Reviews: App
Moco+ - Chat, Meet People                          65124.233794
Calculator                                         66612.233794
Truth or Dare Pro                                  66669.233794
Private Dating, Hide App- Blue for PrivacyHider    66669.233794
Ad Blocker for SayHi                               66665.233794
                                                       ...     
FP Markets                                         66668.233794
FP Boss                                            66668.233794
FP FCU                                             66621.233794
Fast Tract Diet                                    66634.233794
Gold Quote - Gold.fr                               66573.233794
Name: Reviews, Length: 941, dtype: float64


In [14]:
#Desviación con respecto a la media RATING
desv_media_rating=np.abs(rating - media_rating)
print("Desviación respecto a la media sobre Rating :",desv_media_rating)

Desviación respecto a la media sobre Rating : App
Moco+ - Chat, Meet People                          0.053742
Calculator                                         1.546258
Truth or Dare Pro                                       NaN
Private Dating, Hide App- Blue for PrivacyHider         NaN
Ad Blocker for SayHi                                    NaN
                                                     ...   
FP Markets                                              NaN
FP Boss                                                 NaN
FP FCU                                             0.546258
Fast Tract Diet                                    0.253742
Gold Quote - Gold.fr                                    NaN
Name: Rating, Length: 941, dtype: float64


>Varianza

In [15]:
varianza_reviews =np.var(reviews)
print("Varianza de reviews:",varianza_reviews)

Varianza de reviews: 57956018239.830574


In [16]:
varianza_rating =np.var(rating)
print("Varianza de rating:",varianza_rating)

Varianza de rating: 0.4098479581467124


>Desviación típica

In [17]:
desviacion_tipica_reviews= np.std(reviews)
print("Desviación tipica de reviews:",desviacion_tipica_reviews)

Desviación tipica de reviews: 240740.56209918298


In [18]:
desviacion_tipica_rating= np.std(rating)
print("Desviación tipica de rating:",desviacion_tipica_rating)

Desviación tipica de rating: 0.640193687993495


>Moda

In [19]:
moda,frecuencia_reviews= stats.mode(reviews)
print("Moda y frecuencia de reviews",moda ,frecuencia_reviews)

Moda y frecuencia de reviews [0] [58]


In [20]:
moda,frecuencia_rating= stats.mode(rating)
print("Moda y frecuencia de rating",moda ,frecuencia_rating)

Moda y frecuencia de rating [nan] [126]


>Mediana

In [21]:
mediana_reviews= np.median(reviews)
print("Mediana de reviews es:",mediana_reviews)

Mediana de reviews es: 1601.0


In [22]:
mediana_rating= np.median(rating)
print("Mediana de rating es:", mediana_rating)

Mediana de rating es: nan


>Media ponderada

In [23]:
w = np.ones(941, dtype=np.complex64)
avg_reviews= np.average(reviews,weights=w)
print("La media ponderada de reviews es:",avg_reviews)

La media ponderada de reviews es: (66669.23379383633+0j)


In [24]:
w = np.ones(941, dtype=np.complex64)
avg_rating= np.average(rating,weights=w)
print("La media ponderada de rating es:",avg_rating)

La media ponderada de rating es: (nan+nanj)


>Robustez

La robustez  se refiere a que el estadístico, no se vea afectado en gran medida por valores atípicos. La mediana es una medida robusta de tendencia central, mientras que el promedio no, puesto que una única medición muy grande o muy pequeña altera significativamente el promedio.

>Coeficiente de variación

In [26]:
coef_var_reviews= np.std(reviews)/np.abs(np.mean(reviews))
print("Coeficiente de variación de reviews:",coef_var_reviews)


Coeficiente de variación de reviews: 3.6109693842235178


In [27]:
coef_var_rating= np.std(rating)/np.abs(np.mean(rating))
print("Coeficiente de variación de rating:",coef_var_rating)

Coeficiente de variación de rating: 0.1544027745367834


**3. Interpretación de los resultados.**
- **Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo incapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.**

Tenino en cuenta las app evaluadas

In [30]:
df.head(2)

Category  Rating  \
App                                                                      
Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
Coloring book moana                             ART_AND_DESIGN     3.9   

                                               Reviews Size  Installs  Type  \
App                                                                           
Photo Editor & Candy Camera & Grid & ScrapBook     159  19M   10,000+  Free   
Coloring book moana                                967  14M  500,000+  Free   

                                               Price Content Rating  \
App                                                                   
Photo Editor & Candy Camera & Grid & ScrapBook     0       Everyone   
Coloring book moana                                0       Everyone   

                                                                   Genres  \
App                                                                         
Photo Editor & Candy Camera & Grid & ScrapBook               Art & Design   
Coloring book moana                             Art & Design;Pretend Play   

                                                    Last Updated Current Ver  \
App                                                                            
Photo Editor & Candy Camera & Grid & ScrapBook   January 7, 2018       1.0.0   
Coloring book moana                             January 15, 2018       2.0.0   

                                                 Android Ver  
App                                                           
Photo Editor & Candy Camera & Grid & ScrapBook  4.0.3 and up  
Coloring book moana                             4.0.3 and up

In [32]:
df.shape

(10841, 12)

Hemos evaluado 10.840 aplicaciones, de las cuales la media de reviews es de 66.669. Analizando la varianza encontramos que los datos de review de las aplicaciones son dispersos estan muy alejados de la media. La mediana que es el valor central de nuestros datos es de 1601 y es muy distinta a la media que es de 66669.

In [28]:
#Resumen datos reviews

print("Datos reviews:")

print("Media de reviews:",media_reviews)
print("Desviación respecto a la media sobre reviews:",desv_media_reviews)
print("Varianza de reviews:",varianza_reviews)
print("Desviación tipica de reviews:",desviacion_tipica_reviews)
print("Moda y frecuencia de reviews",moda ,frecuencia_reviews)
print("Mediana de reviews es:",mediana_reviews)
print("La media ponderada de reviews es:",avg_reviews)
print("Coeficiente de variación de reviews:",coef_var_reviews)


Datos reviews:
Media de reviews: 66669.23379383635
Desviación respecto a la media sobre reviews: App
Moco+ - Chat, Meet People                          65124.233794
Calculator                                         66612.233794
Truth or Dare Pro                                  66669.233794
Private Dating, Hide App- Blue for PrivacyHider    66669.233794
Ad Blocker for SayHi                               66665.233794
                                                       ...     
FP Markets                                         66668.233794
FP Boss                                            66668.233794
FP FCU                                             66621.233794
Fast Tract Diet                                    66634.233794
Gold Quote - Gold.fr                               66573.233794
Name: Reviews, Length: 941, dtype: float64
Varianza de reviews: 57956018239.830574
Desviación tipica de reviews: 240740.56209918298
Moda y frecuencia de reviews [nan] [58]
Mediana de reviews es: 

In [29]:
#Resumen datos rating

print("Datos rating:")

print("Varianza de rating:",varianza_rating)
print("Desviación respecto a la media sobre Rating :",desv_media_rating)
print("Desviación tipica de rating:",desviacion_tipica_rating)
print("Moda y frecuencia de rating",moda ,frecuencia_rating)
print("Mediana de rating es:", mediana_rating)
print("La media ponderada de rating es:",avg_rating)
print("Coeficiente de variación de rating:",coef_var_rating)

Datos rating:
Varianza de rating: 0.4098479581467124
Desviación respecto a la media sobre Rating : App
Moco+ - Chat, Meet People                          0.053742
Calculator                                         1.546258
Truth or Dare Pro                                       NaN
Private Dating, Hide App- Blue for PrivacyHider         NaN
Ad Blocker for SayHi                                    NaN
                                                     ...   
FP Markets                                              NaN
FP Boss                                                 NaN
FP FCU                                             0.546258
Fast Tract Diet                                    0.253742
Gold Quote - Gold.fr                                    NaN
Name: Rating, Length: 941, dtype: float64
Desviación tipica de rating: 0.640193687993495
Moda y frecuencia de rating [nan] [126]
Mediana de rating es: nan
La media ponderada de rating es: (nan+nanj)
Coeficiente de variación de rating: 0.